hoot-core/src/main/cpp/hoot/core/cmd/ConflateCmd.cpp

```diff
@@ -184,32 +184,60 @@ int ConflateCmd::runSimple(QStringList args)
   }
   int currentTask = 1;
 
-  // read input 1
-  progress.set(
-    _getJobPercentComplete(currentTask - 1),
-    "Loading reference map: ..." + input1.right(maxFilePrintLength) + "...");
   OsmMapPtr map(new OsmMap());
-  IoUtils::loadMap(map, input1, ConfigOptions().getReaderConflateUseDataSourceIds1(),
-                   Status::Unknown1);
-  currentTask++;
-
   ChangesetProviderPtr pTagChanges;
-  if (isDiffConflate)
+
+  //  Loading order is important if datasource IDs 2 is true but 1 is not
+  if (!ConfigOptions().getReaderConflateUseDataSourceIds1() &&
+       ConfigOptions().getReaderConflateUseDataSourceIds2() &&
+      !isDiffConflate)
```

Comment:
Load the secondary dataset first to preserve those IDs in case of collision in reference.